dyntastic/pydantic_compat.py

```diff
@@ -134,7 +134,14 @@ def collect_valid_fields(cls, v, info):
 
 
 def field_type(model: Type[pydantic.BaseModel], field: str) -> Type:
-    return annotation(model_fields(model)[field])
+    model_field = next(
+        (v for k, v in model_fields(model).items() if field in (k, v.alias)),
+        None,
+    )
+    if model_field:
+        return annotation(model_field)
+
+    raise ValueError(f"Field {field} is not present in {model}")
```

Comment:
What do you think about removing the function entirely, along with the `hash_key_type` validation in `batch_get`? It might be better to just serialize what we can and pass it along to dynamo. (It might cause an empty set of results rather than a clear error, but it's a lot of logic to capture a relatively unlikely scenario).

Reply:
Personally, I like having it there. It feels consistent to have type checking prior to interaction with dynamo in this case as well.

Reply:
Good feedback :+1: 

The behavior of Pydantic is to favor alias when there is one, so we should probably align this function with that (i.e. first check for any fields with a matching alias, then only if there was no match look up the field by name). Really this only matters if someone does something like the snippet below which seems unlikely, but might as well support it since Pydantic does.

```python
class MyTable(Dyntastic):
    __hash_key__ = "id"

    id: id
    actual_id: str = Field(alias="id")  # this is the hash key
```

We should probably raise an error in the case of multiple fields with the same alias as well (though that should be done when subclassing Dyntastic since it more than just batch_get). I can make that change before releasing unless you're feeling ambitious.